Source/WebKit/WebProcess/WebStorage/StorageNamespaceImpl.cpp

```diff
@@ -67,27 +57,14 @@ Ref<StorageNamespaceImpl> StorageNamespaceImpl::createTransientLocalStorageNames
     return adoptRef(*new StorageNamespaceImpl(StorageType::TransientLocal, identifier, std::nullopt, &topLevelOrigin, quotaInBytes));
 }
 
-StorageNamespaceImpl::StorageNamespaceImpl(WebCore::StorageType storageType, Identifier storageNamespaceID, const std::optional<PageIdentifier>& pageIdentifier, WebCore::SecurityOrigin* topLevelOrigin, unsigned quotaInBytes)
+StorageNamespaceImpl::StorageNamespaceImpl(WebCore::StorageType storageType, Identifier storageNamespaceID, const std::optional<PageIdentifier>& pageIdentifier, const WebCore::SecurityOrigin* topLevelOrigin, unsigned quotaInBytes)
     : m_storageType(storageType)
     , m_storageNamespaceID(storageNamespaceID)
     , m_sessionPageID(pageIdentifier)
-    , m_topLevelOrigin(topLevelOrigin)
+    , m_topLevelOrigin(const_cast<WebCore::SecurityOrigin *>(topLevelOrigin))
```

Comment:
Can we make m_topLevelOrigin a. RefPtr<const WebCore::SecurityOrigin> instead of having a const_cast?

Reply:
Done, made it `const`.